subprojects/core/src/main/java/org/gradle/execution/ProjectExecutionServices.java

```diff
@@ -136,15 +137,15 @@ TaskExecuter createTaskExecuter(
         TaskSnapshotter taskSnapshotter,
         WorkExecutor<ExecutionRequestContext, CachingResult> workExecutor
     ) {
-
-        ExecuteActionsTaskExecuter.VfsInvalidationStrategy vfsInvalidationStrategy = VirtualFileSystemServices.isPartialInvalidationEnabled(startParameter)
+        StartParameterInternal startParameterInternal = (StartParameterInternal) startParameter;
```

Comment:
Can we inject `StartParameterInternal` instead of casting it?

Reply:
I tried and we can't :(